lang/php/lib/Datum/AvroIOBinaryEncoder.php

```diff
@@ -187,4 +188,46 @@ public function writeBytes($bytes)
         $this->writeLong(strlen($bytes));
         $this->write($bytes);
     }
+
+    public function writeDecimal($decimal, int $scale, int $precision): void
+    {
+        if (!is_numeric($decimal)) {
+            throw new AvroException('Decimal must be a numeric value');
+        }
+
+        $value = $decimal * (10 ** $scale);
+        if (!is_int($value)) {
+            $value = (int) round($value);
+        }
+        if (abs($value) > (10 ** $precision - 1)) {
+            throw new AvroException('Decimal value is out of range');
+        }
+
+        $packed = pack('J', $value);
```

Comment:
pack('J', ...) is machine-endian and width-dependent
Avro decimal bytes must be big-endian two’s complement